scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,631 @@
+        defining the lower and upper bounds for the optimizing argument of
+        `func`. It is required to have ``len(bounds) == len(x)``.
+        ``len(bounds)`` is used to determine the number of parameters in ``x``.
+    niter : int, optional
+        The number of gensa iterations
+    T : float, optional
+        Initial value for temperature. The higher the initial temperature, the
+        the higher the probability of hill-climbing for the initial iterations.
+    visitparam : float, optional
+        Parameter for visiting distribution. Higher value means the
+        visiting distribution has a heavier tail which makes the
+        algorithm jump to a farer region. The value range is ]0; 3]
+    acceptparam : float, optional
+        Parameter for acceptance distribution. It is used to control the
+        probability of acceptance. The lower the acceptance parameter, the
+        smaller the probability of acceptance. It has to be any negative value.
+    maxcall : int, optional
+        Soft limit for the number of objective function calls. If the
+        algorithm is in the middle of a local search, this number will be
+        exceeded, the algorithm will stop just after the local search is
+        done.
+    args : tuple, optional
+        Any additional fixed parameters needed to
+        completely specify the objective function.
+    seed : int or `np.random.RandomState`, optional
+        If `seed` is not specified the `np.RandomState` singleton is used.
+        If `seed` is an int, a new `np.random.RandomState` instance is used,
+        seeded with seed.
+        If `seed` is already a `np.random.RandomState instance`, then that
+        `np.random.RandomState` instance is used.
+        Specify `seed` for repeatable minimizations. The random numbers
+        generated with this seed only affect the visiting distribution
+        function and new coordinates generation.
+
+    Returns
+    -------
+    res : OptimizeResult
+        The optimization result represented as a ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``fun`` the value
+        of the function at the solution, and ``message`` which describes the
+        cause of the termination.
+        See `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+    GenSA is an implementation of the General Simulated Annealing algorithm
+    (GSA [2]_). This stochastic approach generalizes CSA (Classical Simulated
+    Annealing) and FSA (Fast Simulated Annealing) to find the neighborhood of
+    minima, then calls a local method (lbfgsb) to find their exact value.
+    GenSA can process complicated and high dimension non-linear objective
+    functions with a large number of local minima as described by [6]_.
+
+    GSA uses a distorted Cauchy-Lorentz visiting distribution, with it shape
+    controlled by the parameter :math:`q_{v}`
```

Comment:
Yes. We keep it short in the formula so that it is easier to read and display. Comment added in the formula explanation.